p8e-api/src/main/kotlin/io/provenance/engine/batch/TxErrorReaper.kt

```diff
@@ -46,47 +50,47 @@ class TxErrorReaper(
         }.forEach {
             try {
                 P8eMDC.set(it.transactionHash.value.toTransactionHashes(), clear = true)
-                sc.fetchTx(it.transactionHash.value).let { transactionStatus ->
-                    if (transactionStatus.isErrored()) {
+                transactionQueryService.fetchTransaction(it.transactionHash.value).let { transactionStatus ->
+                    if (transactionStatus.isNotFound()) {
+                        log.warn("Retrying transaction not found in mempool with hash ${it.transactionHash.value}")
+                        transaction { transactionStatusService.retryDead(it, transactionStatus.data) }
+                    } else if (transactionStatus.isErrored()) {
                         transaction { transactionStatusService.setError(it, transactionStatus.getError()) }
-                    } else {
-                        P8eMDC.set(transactionStatus.height.toLong().toBlockHeight())
-                        val blockHeight = transactionStatus.height.toLong()
+                    } else if (transactionStatus.isSuccessful()) {
+                        P8eMDC.set(transactionStatus.height.toBlockHeight())
+                        val blockHeight = transactionStatus.height
                         if (blockHeight <= latestBlockHeight) { // EventStream is past this height, need to process manually
-                            transactionStatus.events()
+                            transactionStatus.scopeEvents()
                                 ?.also { events -> log.error("indexing ${events.size} events missed by eventStream") }
                                 ?.map { event -> event.toScopeEvent() }
                                 ?.also { events -> scopeStream.queueIndexScopes(blockHeight, events) }
                         }
                     }
                 }
-            } catch (e: CosmosRemoteInvocationException) {
-                if (e.status == HttpStatus.NOT_FOUND.value()) {
-                    log.warn("Retrying transaction not found in mempool with hash ${it.transactionHash.value}")
-                    transaction { transactionStatusService.retryDead(it, e.message) }
-                } else {
-                    log.error("Error fetching status for tx hash ${it.transactionHash.value}", e)
-                }
             } catch (t: Throwable) {
                 log.warn("Error processing expired transaction", t)
             }
         }
     }
 
-    private fun TxQuery.isErrored() = (code ?: 0) > 0
+    private fun GetTxResult.isNotFound() = txResult == null
+
+    private fun GetTxResult.isErrored() = txResult?.code != null && txResult?.code > 0
+
+    private fun GetTxResult.isSuccessful() = txResult?.code == 0
```

Comment:
I did an experiment just to confirm but null == 0 returns false, so it will still return false if there is no code, the isErrored function has to have the null check to make the comparison to 0 only happen if there is a code